src/Workspaces/Core/Portable/PatternMatching/AllLowerCamelCaseMatcher.cs

```diff
@@ -97,7 +104,7 @@ private PatternMatchKind GetKind(CamelCaseResult result)
                     }
 
                     var candidateHump = _candidateHumps[humpIndex];
-                    if (char.ToLower(_candidate[candidateHump.Start]) == patternCharacter)
+                    if (ToLower(_candidate[candidateHump.Start], _textInfo) == patternCharacter)
```

Comment:
Incredibly hot path.  And the common case is to be ascii.  this avoids one check+branch in char where it checks if is latin first.